R/LearnerSurvXGBoost.R

```diff
@@ -0,0 +1,158 @@
+      ))
+      # param deps
+      ps$add_dep("print_every_n", "verbose", CondEqual$new(1L))
+      ps$add_dep("sample_type", "booster", CondEqual$new("dart"))
+      ps$add_dep("normalize_type", "booster", CondEqual$new("dart"))
+      ps$add_dep("rate_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("skip_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("one_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("tree_method", "booster", CondAnyOf$new(c("gbtree", "dart")))
+      ps$add_dep("grow_policy", "tree_method", CondEqual$new("hist"))
+      ps$add_dep("max_leaves", "grow_policy", CondEqual$new("lossguide"))
+      ps$add_dep("max_bin", "tree_method", CondEqual$new("hist"))
+      ps$add_dep("sketch_eps", "tree_method", CondEqual$new("approx"))
+      ps$add_dep("feature_selector", "booster", CondEqual$new("gblinear"))
+      ps$add_dep("top_k", "booster", CondEqual$new("gblinear"))
+      ps$add_dep("top_k", "feature_selector", CondAnyOf$new(c("greedy", "thrifty")))
+
+      ps$values = list(nrounds = 1L, verbose = 0L, objective = "survival:cox",
+        eval_metric = "cox-nloglik")
+
+      super$initialize(
+        id            = "surv.xgboost",
+        param_set     = ps,
+        predict_types = c("crank", "lp"),
+        feature_types = c("logical", "integer", "numeric"),
+        properties    = c("weights", "missings", "importance"),
+        packages      = c("xgboost")
+      )
+    },
+
+    #' @description
+    #' The importance scores are calculated with [xgboost::xgb.importance()].
+    #'
+    #' @return Named `numeric()`.
+    importance = function() {
+      if (is.null(self$model)) {
+        stopf("No model stored")
+      }
+
+      imp = xgboost::xgb.importance(
+        feature_names = self$model$features,
+        model = self$model
+      )
+      set_names(imp$Gain, imp$Feature)
+    }
+  ),
+
+  private = list(
+    .train = function(task) {
+
+      pars = self$param_set$get_values(tags = "train")
+      targets = task$target_names
+
+      data   = task$data(cols = task$feature_names)
+      target = task$data(cols = task$target_names)
+      label = target[[targets[1]]]
+      status = target[[targets[2]]]
+      label[status != 1] = -1L * label[status != 1]
+      data = xgboost::xgb.DMatrix(
+        data = data.matrix(data),
+        label = label)
+
+      if ("weights" %in% task$properties) {
+        xgboost::setinfo(data, "weight", task$weights$weight)
+      }
+
+      if (is.null(pars$watchlist)) {
+        pars$watchlist = list(train = data)
+      }
+
+      invoke(xgboost::xgb.train, data = data, .args = pars)
+
+    },
+
+    .predict = function(task) {
+
+      pars     = self$param_set$get_values(tags = "predict")
+      model    = self$model
+      newdata  = data.matrix(task$data(cols = task$feature_names))
+      newdata  = newdata[, model$feature_names, drop = FALSE]
+      crank = invoke(predict, model, newdata = newdata, .args = pars)
```

Comment:
Just to confirm, I assume this is predicting a linear predictor here? Or is it genuinely just some ranking prognostic index with no other meaningful value?

Reply:
yes